From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 8 Jun 2017 17:07:56 +0200
Subject: [PATCH] layout: Make the hot corner optional

Whether people love or hate the hot corner depends in large extents
on hardware sensitivity and habits, which is hard to get right
universally. So bite the bullet and add an option to enable or
disable hot corners ...

https://bugzilla.gnome.org/show_bug.cgi?id=688320
---
 data/org.gnome.shell.gschema.xml.in | 8 ++++++++
 js/ui/layout.js                     | 8 ++++++++
 2 files changed, 16 insertions(+)

diff --git a/data/org.gnome.shell.gschema.xml.in b/data/org.gnome.shell.gschema.xml.in
index 63a1903..8ec1d26 100644
--- a/data/org.gnome.shell.gschema.xml.in
+++ b/data/org.gnome.shell.gschema.xml.in
@@ -38,6 +38,14 @@
         load all extensions regardless of the versions they claim to support.
       </description>
     </key>
+    <key name="enable-hot-corners" type="b">
+      <default>true</default>
+      <summary>Enable hot corners</summary>
+      <description>
+        If true, the overview can be accessed by moving the mouse to the
+        top-left corner.
+      </description>
+    </key>
     <key name="favorite-apps" type="as">
       <default>[ 'epiphany.desktop', 'evolution.desktop', 'rhythmbox.desktop', 'shotwell.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]</default>
       <summary>List of desktop file IDs for favorite applications</summary>
diff --git a/js/ui/layout.js b/js/ui/layout.js
index 6f81039..3d53bd4 100644
--- a/js/ui/layout.js
+++ b/js/ui/layout.js
@@ -268,6 +268,9 @@ var LayoutManager = new Lang.Class({
         this._backgroundGroup.lower_bottom();
         this._bgManagers = [];
 
+        global.settings.connect('changed::enable-hot-corners',
+                                Lang.bind(this, this._updateHotCorners));
+
         // Need to update struts on new workspaces when they are added
         global.screen.connect('notify::n-workspaces',
                               this._queueUpdateRegions.bind(this));
@@ -364,6 +367,11 @@ var LayoutManager = new Lang.Class({
         });
         this.hotCorners = [];
 
+        if (!global.settings.get_boolean('enable-hot-corners')) {
+            this.emit('hot-corners-changed');
+            return;
+        }
+
         let size = this.panelBox.height;
 
         // build new hot corners
